doc/contributing/maintaining-shared-library-support.md

```diff
@@ -0,0 +1,121 @@
+# Maintaining shared library support
+
+Node.js unofficially supports a build option where Node.js is built as
+a shared library. The shared library is node.dll on windows, and
+libnode on other platforms.
```

Comment:
As of 88beaf01f14fb042de72dfbbc15f80c71b33bc9e the library is called `libnode.dll` on Windows (not `node.dll`).